addons/event_sale/tests/test_event_sale.py

```diff
@@ -279,3 +280,22 @@ def test_cancel_so(self):
         self.assertEqual(event.seats_expected, 1)
         self.sale_order.action_cancel()
         self.assertEqual(event.seats_expected, 0)
+
+    def test_check_event_id_and_event_ticket_id(self):
+        so_1 = self.env['sale.order'].create({
+            'partner_id': self.event_customer.id,
+        })
+
+        sol_1 = (self.env['sale.order.line'].create({
+            'name': self.event_0.name,
+            'product_id': self.event_product.id,
+            'product_uom_qty': 1,
+            'product_uom': self.event_product.uom_id.id,
+            'price_unit': self.event_product.list_price,
+            'order_id': so_1.id,
+            'event_id': self.event_0.id,
+            'event_ticket_id': self.ticket.id,
+        }))
+
+        with self.assertRaises(UserError, msg="Event and Event Ticket are required!"):
+            sol_1.write({'event_id': False, 'event_ticket_id': False})
```

Comment:
Here you just have to add the confirmation. This way we actually get the original error if we remove your exception purposefully for some reason :)

And it reflects the reproduction steps fully
```suggestion
        with self.assertRaises(UserError, msg="Event and Event Ticket are required!"):
            sol_1.write({'event_id': False, 'event_ticket_id': False})
        so_1.action_confirm()
```